pkg/controllers/dynakube/extension/otel/statefulset_test.go

```diff
@@ -0,0 +1,308 @@
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			},
+		}
+
+		statefulSet := getStatefulset(t, dk)
+
+		assert.Equal(t, dk.Spec.Templates.OpenTelemetryCollector.Resources, statefulSet.Spec.Template.Spec.Containers[0].Resources)
+	})
+}
+
+func TestLabels(t *testing.T) {
+	t.Run("the default labels", func(t *testing.T) {
+		dk := getTestDynakube()
+
+		statefulSet := getStatefulset(t, dk)
+
+		appLabels := buildAppLabels(dk.Name)
+
+		assert.Equal(t, appLabels.BuildLabels(), statefulSet.ObjectMeta.Labels)
+		assert.Equal(t, &metav1.LabelSelector{MatchLabels: appLabels.BuildMatchLabels()}, statefulSet.Spec.Selector)
+		assert.Equal(t, appLabels.BuildLabels(), statefulSet.Spec.Template.ObjectMeta.Labels)
+	})
+
+	t.Run("custom labels", func(t *testing.T) {
+		dk := getTestDynakube()
+		customLabels := map[string]string{
+			"a": "b",
+		}
+		dk.Spec.Templates.OpenTelemetryCollector.Labels = customLabels
+
+		statefulSet := getStatefulset(t, dk)
+
+		appLabels := buildAppLabels(dk.Name)
+		podLabels := maputils.MergeMap(customLabels, appLabels.BuildLabels())
+
+		assert.Equal(t, appLabels.BuildLabels(), statefulSet.ObjectMeta.Labels)
+		assert.Equal(t, &metav1.LabelSelector{MatchLabels: appLabels.BuildMatchLabels()}, statefulSet.Spec.Selector)
+		assert.Equal(t, podLabels, statefulSet.Spec.Template.ObjectMeta.Labels)
+	})
+}
+
+func TestAnnotations(t *testing.T) {
+	t.Run("the default annotations", func(t *testing.T) {
+		statefulSet := getStatefulset(t, getTestDynakube())
+
+		assert.Len(t, statefulSet.ObjectMeta.Annotations, 1)
+		assert.Empty(t, statefulSet.Spec.Template.ObjectMeta.Annotations)
+	})
+
+	t.Run("custom annotations", func(t *testing.T) {
+		dk := getTestDynakube()
+		customAnnotations := map[string]string{
+			"a": "b",
+		}
+		dk.Spec.Templates.OpenTelemetryCollector.Annotations = customAnnotations
+
+		statefulSet := getStatefulset(t, dk)
+
+		assert.Len(t, statefulSet.ObjectMeta.Annotations, 1)
+		assert.Empty(t, statefulSet.ObjectMeta.Annotations["a"])
+		assert.Len(t, statefulSet.Spec.Template.ObjectMeta.Annotations, 1)
+		assert.Equal(t, customAnnotations, statefulSet.Spec.Template.ObjectMeta.Annotations)
+	})
+}
+
+func TestTolerations(t *testing.T) {
+	t.Run("the default tolerations", func(t *testing.T) {
+		statefulSet := getStatefulset(t, getTestDynakube())
+
+		assert.Empty(t, statefulSet.Spec.Template.Spec.Tolerations)
+	})
+
+	t.Run("custom tolerations", func(t *testing.T) {
+		dk := getTestDynakube()
+
+		customTolerations := []corev1.Toleration{
+			{
+				Key:      "a",
+				Operator: corev1.TolerationOpEqual,
+				Value:    "b",
+				Effect:   corev1.TaintEffectNoSchedule,
+			},
+		}
+		dk.Spec.Templates.OpenTelemetryCollector.Tolerations = customTolerations
+
+		statefulSet := getStatefulset(t, dk)
+
+		assert.Equal(t, customTolerations, statefulSet.Spec.Template.Spec.Tolerations)
+	})
+}
```

Comment:
Please add TestSecurityContext and TestUpdateStrategy functions to make sure we do set them.

Reply:
fixed in 1767f67d